### ISSUE TITLE
about GPU usage information and solver.prototxt setting

### ISSUE THREAD

**Issue by 401qingkong** [NVIDIA/DIGITS#901]
hi,I have two questions:
first: I install digits version 3,but i can't found the GPU usage information when i train a model. where do i need to set it
second:where can i set and adjust the parameters in the solver.prototxt file,for example,weight decay in neural network.  Now it generates automatically when i begin to train a model.
  thanks in advance!

**Comment by enoonIT** [comment]
I recently formatted the PC and now I now longer get the GPU information.
The weird thing is that deviceQuery works and I get the information on the available number of GPUs.

**Comment by 401qingkong** [comment]
@gheinrich  thanks for your reply. I can see the picture on the home page,but i can't see the detailed GPU usage information. And i think it's better to set the solver  parameters by ourselves. it can help us modify the model.

**Comment by 401qingkong** [comment]
@enoonIT  Me too, i  can only see the  the available number of GPUs.

**Comment by enoonIT** [comment]
I think exposing solver parameters would be extremely useful.
Also the current way of initializing the weight decay (LR divided by a constant) is confusing and makes experimenting with different solvers awkward (you change the starting LR and my magic also change the WD...)

**Comment by IsaacYangSLA** [comment]
@401qingkong & @enoonIT , can you try nvidia-smi to see if you can get GPU status?  It's a small utility and you can run it from your shell.  Enumerating GPUs and retrieving their status are done by two different tools, so it's possible that one works while the other has some issues.

**Comment by enoonIT** [comment]
@IsaacYangSLA: nvidia-smi works as expected, GPU is correctly detected and temperature is reported

**Comment by IsaacYangSLA** [comment]
@enoonIT , thanks for the info.  That told us the nvml library is working.  Can you try the python script, device_query.py, inside your DIGITS/digits folder?  It calls cudart to find devices and their properties, and calls nvml to get GPU status.  Basically, we want to make sure [this line](https://github.com/NVIDIA/DIGITS/blob/master/digits/device_query.py#L295) returns the status.  You can add some printout inside that call and check where it fails, in case that call returns None.

**Comment by enoonIT** [comment]
This is the output from device query:

```
Device #0:
>>> CUDA attributes:
  name                         GeForce GTX TITAN X
  totalGlobalMem               12884180992
  clockRate                    1076000
  major                        5
  minor                        2
>>> NVML attributes:
  Total memory                 12287 MB
  Used memory                  3867 MB
  Memory utilization           6%
  GPU utilization              22%
  Temperature                  71 C

```

One thing to note is that I'm using the bvlc caffe, not the nvidia version

**Comment by JieZou1** [comment]
@enoonIT @IsaacYangSLA 
I am facing the similar (maybe worse) situation: no GPU info is displayed on the DIGITS GUI. Following the discussion aboves, below are the facts in my case:

I am running DIGITS on Windows.

On DIGITS Home GUI, the GPU enumerating message is shown and correct.
With NVidia-smi, the GPU info seems listing correctly (copied below). The temperature is indeed going up when I am training a model, and then dropping down when training is finished.

Wed Jul 20 12:28:39 2016
+------------------------------------------------------+
| NVIDIA-SMI 353.90     Driver Version: 353.90         |
|-------------------------------+----------------------+----------------------+
| GPU  Name            TCC/WDDM | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 770    WDDM  | 0000:01:00.0     N/A |                  N/A |
| 31%   54C    P0    N/A /  N/A |    793MiB /  4096MiB |     N/A      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID  Type  Process name                               Usage      |
|=============================================================================|
|    0                  Not Supported                                         |
+-----------------------------------------------------------------------------+

However, device_query.py doesn't print NVML attributes. I got CUDA attributes only, copied below:
Device #0:
>>> CUDA attributes:
  name                         GeForce GTX 770
  totalGlobalMem               4294967296
  clockRate                    1189000
  major                        3
  minor                        0

Does that mean that the GPU is actually not used during the training, or there are some issues in device_query.py when calling NVML? Thanks.

**Comment by IsaacYangSLA** [comment]
@JieZou1 , there is limitation on nvml due to Windows and the driver.  Therefore, DIGITS on Windows only reports # of GPU, and no information about temp/memory/utilization.
What you mentioned about temp rising and dropping from nvml told us your training was running on GPU.  Can you also check your memory usage during training?  If you make batch very large in train_val.prototxt and see GPU memory allocation error, you can be pretty sure it's running on GPU.

**Comment by IsaacYangSLA** [comment]
@JieZou1 , any updates?

**Comment by JieZou1** [comment]
@IsaacYangSLA I agree with you that I believe the program is running on GPU. As to why no GPU info displayed, I accepted your suggestion and I did not do any further investigation. Sorry.

**Comment by IsaacYangSLA** [comment]
@JieZou1, thanks for the reply.  I will close this one since it seems the issue is the limitation of Windows, not DIGITS.
